networks/twatter.py

```diff
@@ -76,61 +76,59 @@ class Twitter(Network):
 
     @taskify
     async def check_tweets(self):
-        """ Parse our stupid homebrew conf file """
 
+        info('checking tweets')
         if not self._twitter_conf:
             warning('_twitter_conf not (yet?) loaded!')
-
             return
 
-        for serv in self._twitter_conf.keys():
-            debug(serv)
+        info('parsing twitter_conf')
+        tooters = self._twitter_conf
 
-            for chann in self._twitter_conf[serv].keys():
-                debug(chann)
+        for tooter in tooters:
+            tweets = self._twitter.get_user_timeline(screen_name = tooter.screen_name)
+            tweets.reverse()
 
-                for tooter in self._twitter_conf[serv][chann]:
+            last_tweet = tooter.last_tweet_id
 
-                    tweets = self._twitter.get_user_timeline(screen_name = tooter['screen_name'])
-                    tweets.reverse()
+            for tweet in tweets:
+                if last_tweet >= 0 and tweet['id'] > last_tweet:
+                    continue
 
-                    # this will be the first tweet in the channel
-                    if tooter['last'] == 1:
-                        tweets = [tweets[-1],]
+                info('new tweet from {}'.format(tweet['user']['screen_name']))
+                tooter.last_tweet_id = tweet['id']
+                tooter.save()
 
+                if tweet['in_reply_to_status_id']:
+                    # don't show tweets that are replies to other users
+                    continue
 
-                    for tweet in tweets:
-                        if tooter['last'] > 1 and tweet['id'] <= tooter['last']:
-                            continue
+                tweet = await self.parse(tweet)
 
-                        info('new tweet from {}'.format(tweet['user']['screen_name']))
+                for channel in tooter.channels:
 
-                        tooter['last'] = tweet['id']
+                    destination = (self.application.Discord.client
+                                   .servers[channel.server.name]
+                                   .channels[channel.name])
 
-                        if tweet['in_reply_to_status_id']:
-                            # don't show tweets that are replies to other users
-                            continue
+                    ''' might be a good idea to add retweet id to each channel'''
```

Comment:
If j4ne goes down or something happens during retweeting I thought it might be a good idea to know what has been retweeted so there wouldn't be any duplicates. Dunno if it is an issue that needs to be considered, though.